url_matcher/matcher.py

```diff
@@ -36,6 +36,10 @@ def is_universal_pattern(self) -> bool:
     def get_includes_for(self, domain: str) -> List[str]:
         return [pattern for pattern in self.include if get_pattern_domain(pattern) == domain]
 
+    def __hash__(self):
+        """Allows unique Patterns to be easily identified and deduped."""
+        return hash((tuple(self.include), tuple(self.exclude), self.priority))
```

Comment:
This implementation makes sense, but the hash may change over the lifetime of Pattern. For example, if you assign a different priority, or extend include or exclude pattern, hash would change, which may cause various strange issues - e.g. you may not be able to retrieve a value from the dict when Pattern is used as a key.

Do you know if we can make Pattern frozen/immutable? Use something like dataclass(frozen=True), and switch from lists to tuples for include/exclude? This could make dataclasses generate `__hash__` method, and it'd avoid the issue with hashes changing over object lifetime. But it'd change how Pattern can be used.

Reply:
Cheers for your good points @kmike ! Updated this on https://github.com/zytedata/url-matcher/pull/4/commits/988e3d97ed7d07d2123fbf60467f63fa0904b857.

Will also be updating `web-poet` and `scrapy-poet` after this has been released, since their **mypy** checks would break.